import ServerError from './ServerError';

export default async function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }
  const error = new ServerError(response.statusText, response.status);

  try {
    error.response = await response.json();
    error.message = error.response.message;
    error.description = error.response.description;
  } catch (e) {
    error.response = {
      message: response.statusText,
    };
  }

  return Promise.reject(error);
}

export function checkError(err) {
  const error = new ServerError(err.message, 0);
  const message = 'The server is not reachable. The server may be down or your internet settings may be down.';
  error.message = message;
  error.description = message;
  error.response = { message };
  return Promise.reject(error);
}
